packages/aws-cdk-lib/core/lib/nodejs-aspect.ts

```diff
@@ -0,0 +1,138 @@
+    const node = provider as cdk.CustomResourceProviderBase;
+    // only replace for nodejs case
+    if (this.isValidRuntime(node.runtime)) {
+      const targetnode = node.node.children.find((child) => child instanceof cdk.CfnResource &&
+      child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      targetnode.addPropertyOverride('Runtime', 'nodejs20.x');
+    }
+  }
+
+  private handleProvider(provider: Provider): void {
+    const functionNodes = provider.node.children.filter((child) => child instanceof lambda.Function);
+    for ( var node of functionNodes) {
+      const targetNode = node.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        targetNode.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+
+      // onEvent Handlers
+      const onEventHandler = provider.onEventHandler as lambda.Function;
+      const onEventHandlerRuntime = onEventHandler.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        onEventHandlerRuntime.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+
+      //isComplete Handlers
+      // Handlers
+      const isCompleteHandler = provider.isCompleteHandler as lambda.Function;
+      const isCompleteHandlerRuntime = isCompleteHandler.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        isCompleteHandlerRuntime.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+    }
+  }
+
+  private handleReceiptRuleSet(ruleSet: ReceiptRuleSet): void {
+    const functionnode = ruleSet.node.findChild('DropSpam');
+    const ses = functionnode.node.findChild('Function') as lambda.CfnFunction;
+    if (ses.runtime && this.isValidRuntime(ses.runtime)) {
+      ses.addPropertyOverride('Runtime', 'nodejs20.x');
+    }
+  }
+
+  /**
+ *Runtime Validation
+ *
+ */
+  private isValidRuntime(runtime: string) : boolean {
+    return runtime === this.targetRuntime;
+  }
+
+  private getRuntimeProperty(node: cdk.CfnResource) {
+    return (node.getResourceProperty('runtime') || node.getResourceProperty('Runtime')) as string;
+  }
+}
+//}
+/**
+ * RuntimeAspect
+ */
+export class RuntimeAspect extends RuntimeAspectsBase {
+
+  /**
+   * DEPRECATED: add tags to the node of a construct and all its the taggable children
+   *
+   * @deprecated use `Tags.of(scope).add()`
+   */
+  public static add(scope: Construct, key: string) {
+    Runtime.of(scope).add(key);
+  }
+
+  constructor(key: string) {
+    super(key);
+  }
+}
+
+/**
+ * Runtime
+ */
+export class Runtime {
```

Comment:
This class isn't really a `Runtime`. You're building an abstraction on top of an `Aspect`, but I think this is going to be confusing as it is. People should be able to use the aspect with the `Aspects` abstraction. Also a `Runtime` in core doesn't really make sense. `Runtimes` should be attached to resources, i.e., Lambda functions or synthetics canaries. Let's remove this and just let people use `Aspects.of(...).add(...)`